internal/webhook/datasciencecluster/validating.go

```diff
@@ -0,0 +1,81 @@
+//go:build !nowebhook
+
+package datasciencecluster
+
+import (
+	"context"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/webhook/shared"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+)
+
+//+kubebuilder:webhook:path=/validate-datasciencecluster,mutating=false,failurePolicy=fail,sideEffects=None,groups=datasciencecluster.opendatahub.io,resources=datascienceclusters,verbs=create;delete,versions=v1,name=datasciencecluster-validator.opendatahub.io,admissionReviewVersions=v1
+//nolint:lll
+
+// Validator implements webhook.AdmissionHandler for DataScienceCluster validation webhooks.
+// It enforces singleton creation rules for DataScienceCluster resources and always allows their deletion.
+type Validator struct {
+	Client  client.Client
+	Decoder admission.Decoder
+	Name    string
+}
+
+// Assert that Validator implements admission.Handler interface.
+var _ admission.Handler = &Validator{}
+
+// InjectDecoder implements admission.DecoderInjector so the manager can inject the decoder automatically.
+func (v *Validator) InjectDecoder(d admission.Decoder) error {
+	v.Decoder = d
+	return nil
+}
+
+// SetupWithManager registers the validating webhook with the provided controller-runtime manager.
+// It configures the webhook to handle DataScienceCluster resources and uses a custom log constructor for context.
+func (v *Validator) SetupWithManager(mgr ctrl.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register("/validate-datasciencecluster", &webhook.Admission{
+		Handler:        v,
+		LogConstructor: shared.NewLogConstructor(v.Name),
+	})
+	// No error to return currently, but return nil for future extensibility
+	return nil
+}
+
+// validateDupCreation denies creation if another instance of the same kind already exists (singleton enforcement).
+func (v *Validator) validateDupCreation(ctx context.Context, req admission.Request) admission.Response {
+	log := logf.FromContext(ctx)
+	return shared.ValidateDupCreation(ctx, v.Client, &req, gvk.DataScienceCluster.Kind, log)
+}
+
+// Handle processes admission requests for create and delete operations on DataScienceCluster resources.
+// It enforces singleton and deletion rules, allowing other operations by default.
+func (v *Validator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	log := logf.FromContext(ctx).WithName(v.Name).WithValues("operation", req.Operation)
```

Comment:
it seems the `v.Name` is added twice to the log:
- here with `WithName`
- in `NewLogConstructor` with `WithValues("webhook", name)` where `name` is actually `v.Name`